tokio/src/runtime/dump.rs

```diff
@@ -30,14 +30,167 @@ pub struct Task {
+    ip: *mut std::ffi::c_void,
+    symbol_address: *mut std::ffi::c_void,
+    symbols: Vec<BacktraceSymbol>,
+}
+
+impl BacktraceFrame {
+    pub(crate) fn from_resolved_backtrace_frame(frame: &backtrace::BacktraceFrame) -> Self {
+        Self {
+            ip: frame.ip(),
+            symbol_address: frame.symbol_address(),
+            symbols: frame
+                .symbols()
+                .iter()
+                .map(BacktraceSymbol::from_backtrace_symbol)
+                .collect(),
+        }
+    }
+
+    /// Return the instruction pointer of this frame.
+    ///
+    /// See the ABI docs for your platform for the exact meaning.
+    pub fn ip(&self) -> *mut std::ffi::c_void {
+        self.ip
+    }
+
+    /// Returns the starting symbol address of the frame of this function.
+    pub fn symbol_address(&self) -> *mut std::ffi::c_void {
+        self.symbol_address
+    }
+
+    /// Return an iterator over the symbols of this backtrace frame.
+    ///
+    /// Due to inlining, it is possible for there to be multiple [BacktraceSymbol] items relating
+    /// to a single frame. The first symbol listed is the "innermost function",
+    /// whereas the last symbol is the outermost (last caller).
+    pub fn symbols(&self) -> impl Iterator<Item = &BacktraceSymbol> {
+        self.symbols.iter()
+    }
+}
+
 /// An execution trace of a task's last poll.
 ///
+/// <div class="warning">
+/// Resolving a backtrace, either via the [`Display`][std::fmt::Display] impl or via
+/// [`resolve_backtraces`][Trace::resolve_backtraces], parses debuginfo, which is
+/// possibly a CPU-expensive operation that can take a platform-specific but
+/// long time to run - often over 100 milliseconds, especially if the current
+/// process's binary is big. In some cases, the platform might internally cache some of the
+/// debuginfo, so successive calls to `resolve_backtraces` might be faster than
+/// the first call, but all guarantees are platform-dependent.
+///
+/// To avoid blocking the runtime, it is recommended
+/// that you resolve backtraces inside of a [spawn_blocking()][crate::task::spawn_blocking]
+/// and to have some concurrency-limiting mechanism to avoid unexpected performance impact.
+/// </div>
+///
 /// See [Handle::dump][crate::runtime::Handle::dump].
 #[derive(Debug)]
 pub struct Trace {
     inner: super::task::trace::Trace,
 }
 
+impl Trace {
+    /// Resolve and return a list of backtraces that are involved in polls in this trace.
+    ///
+    /// The exact backtraces included here are unstable and might change in the future,
+    /// but you can expect one backtrace (one [`Vec<BacktraceFrame>`]) for every call to
+    /// [`poll`] to a bottom-level Tokio future - so if something like [`join!`] is
+    /// used, there will be a backtrace for each future in the join.
+    ///
+
+    /// [`poll`]: std::future::Future::poll
+    /// [`join!`]: macro@join
+    pub fn resolve_backtraces(&self) -> Vec<Vec<BacktraceFrame>> {
```

Comment:
I'm not sure the best shape of the api here ... but I think defining a custom `Backtrace` and returning a `Vec<Backtrace>` is more intuitive than just returning a `Vec<Vec<_>>`. 

```rust
pub struct Backtrace {
    inner: Vec<BacktraceFrame>,
}
impl Backtrace {
    fn frames(&self) -> impl Iterator<Item = &BacktraceFrame> { ... }
}
pub fn resolve_backtraces(&self) -> Vec<Backtrace> { ... }
```

While this would involve adding a new public struct, I don't think this would be harmful so much.

What do you think? (It would also be nice to be able to hear @Darksonn's opinion as well.)

Reply:
> What do you think?

I don't care either way

Reply:
Seems reasonable enough to me to have an extra struct.

Reply:
The new version has it